### PR TITLE
feat: add deterministic conversational search UX

### LINKED ISSUE #23: Add conversational search UX for global and guide-scoped search without AI
## Summary
Add lightweight conversational search UX without AI across both global and within-guide search. Use the shared search state to produce helpful templated responses, suggested refinements, and follow-up filter chips while staying fully static.

## Dependencies
Depends on:

- #20 for the shared scoped search foundation.
- The global search UI ticket for cross-guide result rendering.
- The within-guide search ticket for guide-scoped result rendering.

## Product Direction
This sits between the deterministic search foundation and any model-backed Worker. It should improve perceived intelligence without adding infrastructure, secrets, model cost, or backend latency.

## Context
A user query like `find me cafes in new york with good vibes` does not require an LLM if the search layer can parse location/category/vibe intent. The UI can summarize the interpreted intent, show ranked results, and offer natural follow-up refinements using deterministic state. The same pattern should work globally and inside a guide.

## Implementation Plan
1. Build on the parsed query and scoring output from #20. Consume structured search state such as `scope`, `location`, `category`, `vibes`, `unmatchedTerms`, `resultCount`, and `topSignals`.
2. Add templated response generation in the browser for global search. Example: `I found 6 cafe picks in New York. The strongest matches are cozy/design-forward spots with notes attached.`
3. Add templated response generation in guide scope. Example: `In this Tokyo guide, I found 4 quiet coffee spots.`
4. Add deterministic follow-up suggestions based on result distribution. Examples: `Quiet`, `Lively`, `Date night`, `Laptop friendly`, `Top picks only`, `Nearby neighborhoods`.
5. Make follow-up chips scope-aware. In global search, chips can refine location/category/vibes across guides. In guide search, chips should stay inside the current guide.
6. Add empty-state guidance. If there are no matches, explain what was understood and offer broadening chips such as `Any category`, `Any vibe`, or nearby guide/city suggestions if available.
7. Preserve transparent behavior. Avoid pretending the app is a human or model-backed assistant. The copy should feel helpful, but results must remain inspectable and deterministic.
8. Add URL state for shareability where practical, including query, selected refinements, and scope.
9. Add tests for response templates, scope-aware follow-up suggestions, chip behavior, empty states, and URL state.

## Example UX
Global query:

```txt
cafes in new york with good vibes
```

The page shows:

```txt
I found 6 cafe picks in New York. The strongest matches are cozy, design-forward, or local-favorite spots.
```

Guide-scoped query inside Tokyo:

```txt
quiet coffee
```

The page shows:

```txt
In this Tokyo guide, I found 4 quiet coffee spots.
```

Suggested refinements:

```txt
Quiet
Lively
Date night
Laptop friendly
Top picks only
```

## Acceptance Criteria
- Conversational summaries run entirely client-side.
- The UX works for both cross-guide search and within-guide search.
- Follow-up chips refine deterministic search state and respect the current search scope.
- Empty states explain what was interpreted and offer useful broadening actions.
- Query/refinement/scope state is shareable via URL where practical.
- Existing guide filters continue to work.
- `pnpm run check`, `pnpm run build`, and relevant frontend tests pass.

## Notes
This ticket should be completed before deciding whether the optional Worker AI layer is worth building. It may deliver enough of the chat feel that a model-backed layer is unnecessary for the first public version.

### PR DESCRIPTION
**Pull request by michaelmwu**
## Description

Adds transparent, client-side search guidance to both global and guide-scoped place search.

- Renders concise summaries from the existing parsed category/vibe state.
- Offers result-distribution refinement chips that append deterministic search terms, plus a clear-search broadening action for empty result sets.
- Keeps retrieval and filtering entirely static; the copy explicitly identifies saved place details and tags as the source.
- Keeps guide query/tag refinements shareable through the existing deep-link URL.
- Adds unit and browser coverage.

## Related Issue

Fixes #23

## How Has This Been Tested?

- `bun run test:frontend -- tests/frontend/searchConversation.test.mjs tests/frontend/placeSearch.test.mjs`
- `bun run check`
- `bun run test:e2e -- tests/e2e/home-browser.spec.ts`

---

> [!NOTE]
> **Low Risk**
> UI-only client behavior on top of existing static search; guide URL updates use replaceState without server changes.
> 
> **Overview**
> Adds **client-side “search conversation”** UI on the home global place search and on guide-scoped search: a short summary from parsed categories/vibes/unmatched terms plus **refinement chips** that append terms or clear the query.
> 
> New shared module `search-conversation.js` drives copy and suggestions from existing `place-search` parser state and visible results (no LLM). Home replaces the ad-hoc summary string with this flow; guides show the panel when a query is active and suggestions respect filters already applied.
> 
> Guide search also **syncs `q` and `tag` into the URL** via `history.replaceState` so filtered searches stay shareable. Unit tests cover conversation/suggestion logic; e2e asserts the home summary and suggestion buttons.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 2821edb8841bfa11c4a61ada7eaefdd9069d825b. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added conversational search summaries and refinement suggestions to guide and global search.
  * Search suggestions can append filters or clear the current search.
  * Search URLs now preserve the active query and selected tags for easier sharing and navigation.

* **Bug Fixes**
  * Search summaries and suggestions now reflect only currently visible results.
  * Improved handling of empty queries, unavailable search data, and searches with no matches.

### CODE REVIEW

File: public/scripts/guide-filters.js [hunk 1]
```diff
@@ -1455,6 +1459,31 @@ if (root) {
...
     }
 
+    if (searchConversation && searchConversationSummary && searchConversationSuggestions) {
+      searchConversation.hidden = !query || !searchState;
+      searchConversationSummary.textContent =
+        query && searchState
+          ? buildSearchConversation({
+              count: visibleCards.length,
+              parsed: searchState.parsed,
+              scopeLabel: "this guide",
```
[the remark this answers is not in the document]
Comment (michaelmwu) [reply public/scripts/guide-filters.js]: Addressed: guide summaries now identify active filters as their scope, and suggestions are limited to the active visible result set.

File: public/scripts/home-browser.js [hunk 2]
```diff
@@ -473,34 +516,7 @@ if (root) {
...
-      } else if (
-        searchResultView === "individual" &&
-        visibleResults.length > INDIVIDUAL_RESULT_LIMIT
-      ) {
-        summaryBits.push(`Showing top ${INDIVIDUAL_RESULT_LIMIT} individual matches`);
-      }
-
-      globalResultsSummary.textContent = summaryBits.join(" · ") || "All guides";
-    }
+    renderSearchConversation(query, state, visibleResults);
```
[the remark this answers is not in the document]
Comment (michaelmwu) [reply public/scripts/home-browser.js:524]: Addressed: individual view now states when only the top 24 matches are shown.
